File: server/src/main/java/com/biz360/service/AiResponseFormatter.java
package com.biz360.service;

import com.biz360.dto.AiResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class AiResponseFormatter {

    public AiResponse lowStock(List<?> data) {

        if (data.isEmpty()) {
            return AiResponse.builder()
                    .success(true)
                    .intent("LOW_STOCK")
                    .message("🎉 Great! No products are currently low in stock.")
                    .data(data)
                    .suggestion("You can create a purchase order for new stock")
                    .build();
        }

        return AiResponse.builder()
                .success(true)
                .intent("LOW_STOCK")
                .message("⚠️ Found " + data.size() + " products that need restocking")
                .data(data)
                .suggestion("Create purchase orders for these products")
                .build();
    }

    public AiResponse totalProducts(int count) {

        return AiResponse.builder()
                .success(true)
                .intent("TOTAL_PRODUCTS")
                .message("📦 Total products in system: " + count)
                .data(count)
                .suggestion("You can add new products anytime")
                .build();
    }

    public AiResponse defaultResponse(String intent) {

        return AiResponse.builder()
                .success(true)
                .intent(intent)
                .message("I understood your request as: " + intent)
                .suggestion("This feature will be enhanced soon")
                .build();
    }
}

File: ai-service/app/ai_response_formatter.py
class AIResponseFormatter:

    def format(self, intent: str, data, message: str):

        
        if intent == "LOW_STOCK":

            items = data if isinstance(data, list) else []

            if not items:
                return "All items are sufficiently stocked."

            top = items[0]

            return (
                f"📦 I found {len(items)} low stock items.\n\n"
                f"🔴 Most critical item: {top.get('productName')} "
                f"(Stock: {top.get('baseQuantity')})\n\n"
                f"👉 Recommendation: You should create a Purchase Order for restocking soon."
            )

       
        if intent == "USER_COUNT":

            count = data.get("totalElements", len(data.get("content", [])))

            return (
                f"👥 Total active users in your ERP system: {count}.\n\n"
                f"👉 Suggestion: You can review inactive users for cleanup or audit."
            )

        
        if intent == "SALES_REPORT":

            total = data.get("totalSales", 0)

            return (
                f"💰 Total sales (excluding cancelled invoices): {total}.\n\n"
                f"📊 Insight: You can compare this with previous month for trend analysis."
            )

       
        if intent == "INVOICE_DETAIL":

            if isinstance(data, dict):
                return (
                    f"🧾 Invoice details retrieved successfully.\n\n"
                    f"Customer: {data.get('customerName')}\n"
                    f"Status: {data.get('status')}\n"
                    f"Net Total: {data.get('netTotal')}\n\n"
                    f"👉 You can download or print this invoice from ERP module."
                )

            return "Invoice data not found."

        if intent == "CREATE_PO":

            return (
                "🛒 Purchase Order flow started.\n\n"
                "👉 Please select an item from low stock list to continue."
            )


        return "I understood your request but need more details to assist properly."